ledger/participant-integration-api/src/main/scala/platform/store/backend/h2/H2StorageBackend.scala

```diff
@@ -57,6 +57,25 @@ private[backend] object H2StorageBackend
     ()
   }
 
+  override def resetAll(connection: Connection): Unit = {
```

Comment:
maybe add reuse here?
how about unifying the truncation in common? the one-by-one style seems compatible with all. and the referential integrity treatment with H2 seems to be doable with some set to false;super.reset;set true sequence.